catatom2osm/app.py

```diff
@@ -199,19 +199,23 @@ def get_labels(self, building_gml, part_gml, other_gml):
             self.building.get_labels(
                 other_gml, self.urban_zoning, self.rustic_zoning
             )
-        
-    def split_zoning(self):
-        """Filter zoning using split."""
-        split = QgsVectorLayer(self.options.split, 'zoningsplit', 'ogr')
-
-        self.urban_zoning.remove_outside_features(split)
-        self.rustic_zoning.remove_outside_features(split)
 
     def get_building(self):
         """Merge building, parts and pools"""
         building_gml = self.cat.read("building")
         part_gml = self.cat.read("buildingpart")
         other_gml = self.cat.read("otherconstruction", True)
+
+        if self.options.split:
+            """Filter zoning using split."""
+            split = QgsVectorLayer(self.options.split, 'zoningsplit', 'ogr')
+
+            self.urban_zoning.remove_outside_features(split)
+            self.rustic_zoning.remove_outside_features(split)
+            building_gml.remove_outside_features(split)
+            part_gml.remove_outside_features(split)
+            other_gml.remove_outside_features(split)
+
```

Comment:
Las capas gml son de sólo lectura. Si intentas borrar algo no se borra. El contenido antes y después de borrar es el mismo. No es necesario si construimos self.zone:

            self.zone = [f['label'] for f in self.rustic_zoning.getFeatures()]
            self.zone += [f['label'] for f in self.urban_zoning.getFeatures()]